main/main.go

```diff
@@ -39,159 +45,204 @@ const (
+		errorLogger.Printf("Status reporting error: %+v", err)
+		os.Exit(statusReportingError)
 	}
-	return nil
+	infoLogger.Println("Installation is complete")
+	os.Exit(successfulExecution)
 }
 
-/* 	Open the logfile and configure the loggers that will be used
-*
-*	The main difference between types of loggers is the label (eg INFO) and additional data provided .
- */
-func initLogging() (*os.File, error) {
-	handlerEnv, err := settings.GetHandlerEnvironment()
+func enable() {
+	operation := "enable"
+	infoLogger.Printf("Extension MrSeq: %d, Environment MrSeq: %d", extensionMrSeq, environmentMrSeq)
+	operationLogger.Printf("[Seq Num: %d] [operation: %s]", environmentMrSeq, operation)
+
+	err := status.ReportTransitioning(environmentMrSeq, operation, "enabling in progress")
+	infoLogger.Println("enabling in progress")
 	if err != nil {
-		return nil, errors.Wrap(err, "Failed to open handler environment")
+		errorLogger.Printf("%+v", err)
+		os.Exit(statusReportingError)
 	}
 
-	logfile = path.Join(handlerEnv.HandlerEnvironment.LogFolder, logfileLogName)
+	var publicSettings extensionPublicSettings
+	var protectedSettings extensionPrivateSettings
 
-	file, err := os.OpenFile(logfile, os.O_APPEND|os.O_CREATE|os.O_WRONLY, 0644)
+	err = settings.GetExtensionSettings(environmentMrSeq, &publicSettings, &protectedSettings)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Failed to create/open %s", logfile)
+		status.ReportError(environmentMrSeq, operation, err.Error())
+		errorLogger.Printf("%+v", err)
+		os.Exit(settingsNotFoundError)
 	}
+	infoLogger.Printf("Public Settings: %v \t Protected Settings: %v", publicSettings, protectedSettings)
 
-	//Sample: [2020-08-18T20:29:16.079902Z] [1.0.0.0] [main.go:148] [INFO]: Test1
-	infoLogger = customLogger{log.New(io.MultiWriter(file, os.Stdout), "", 0), infoOperation}
-	warningLogger = customLogger{log.New(io.MultiWriter(file, os.Stderr), "", 0), warningOperation}
-	errorLogger = customLogger{log.New(io.MultiWriter(file, os.Stderr), "", 0), errorOperation}
-
-	return file, nil
-}
+	infoLogger.Printf("Provided Name is: %s", publicSettings.Name)
 
-// extension specific PublicSettings
-type PublicSettings struct {
-	Script   string   `json:"script"`
-	FileURLs []string `json:"fileUris"`
+	err = status.ReportSuccess(environmentMrSeq, operation, "enabling is complete")
+	if err != nil {
+		errorLogger.Printf("%+v", err)
+		os.Exit(statusReportingError)
+	}
+	infoLogger.Println("enabling is complete")
+	os.Exit(successfulExecution)
 }
 
-// extension specific ProtectedSettings
-type ProtectedSettings struct {
-	SecretString       string   `json:"secretString"`
-	SecretScript       string   `json:"secretScript"`
-	FileURLs           []string `json:"fileUris"`
-	StorageAccountName string   `json:"storageAccountName"`
-	StorageAccountKey  string   `json:"storageAccountKey"`
-}
+func disable() {
+	operation := "disable"
+	infoLogger.Printf("Extension MrSeq: %d, Environment MrSeq: %d", extensionMrSeq, environmentMrSeq)
```

Comment:
Most of the code for all these operations are similar (except for maybe enable). Try to maybe create a common utility function for all instead of having so much code duplication to make it easier to read and manage.